### PR TITLE
Add basic DMA example, memory to memory transfer

### PR DESCRIPTION
**Pull request by jlpettersson**
This example is intended to show basic DMA usage, without involving any peripherals.

It is similar to the [hello_dma.c](https://github.com/raspberrypi/pico-examples/blob/master/dma/hello_dma/hello_dma.c) example from official Raspberry Pi.

The structure is taken from the already existing [spi_dma example](https://github.com/rp-rs/rp-hal/blob/main/rp2040-hal/examples/spi_dma.rs).

Also fixing a typo in DMA docs.

### CODE REVIEW

File: rp2040-hal/examples/dma.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+
+    // Slow blink on success
+    loop {
+        led_pin.set_high().unwrap();
+        delay.delay_ms(500);
+        led_pin.set_low().unwrap();
+        delay.delay_ms(500);
+    }
```
Comment (ithinuel) [inline rp2040-hal/examples/dma.rs:90]: You can compare arrays directly like this:

```suggestion
    let delay = if tx_buf == rx_buf {
        500
    } else {
        100
    };
    
    // Slow blink on success
    loop {
        led_pin.set_high().unwrap();
        delay.delay_ms(delay);
        led_pin.set_low().unwrap();
        delay.delay_ms(delay);
    }
```

See [here](https://godbolt.org/z/93zjss7W4) for the output.

**Review by ithinuel — CHANGES_REQUESTED** [review]
It overall looks good to me.

File: rp2040-hal/examples/dma.rs (hunk 1 above)
Comment (9names) [reply rp2040-hal/examples/dma.rs]: I thought I'd check the asm output of this vs the iter().zip way I wrote the on-target-test version on this.
[Lowering to memcmp feels very cheaty](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DEArgoKkl9ZATwDKjdAGFUtEywYgATBtKOAMngMmABy7gBGmMQgAKykAA6oCoR2DC5uHt6%2Bicm2AoHBYSyR0XFWmDapQgRMxATp7p4%2BlpjWeQzVtQQFoRFRsZY1dQ2ZzeZdPUUlsQCUlqgmxMjsHACkXgDMYGCrMQBCDKgA%2BubouwAiqxoAglfX8SbhANRUDC%2BoqBAzTwC0qxvYJ7hD60J6rADsezuTxhT3oBCeTDBG3OYP2ki8JgAHP89lwAGwXXHQ2HwoHI1G7PYaXEEokbKE3WFPNAseK1TAQdb4pikbnhGZ3CGXG5CzZU5BUYAQWrEJgAT0wAEdBTERfdHi83qz2cROUwQGCvIT9tjaSbzqRwobuVSzQy6Wrvn8AUCQRDGddmUj/pcUUCheD1WKNhKpTLiHL5XgsCqiTcHs9XizUGyOTKbca7TiHRarZmTXt7XiLc7/oDga4PSSYfxiE88A23hoAHQtpgt%2BgML5gyE15kNqiIql4C5Pba%2BgP7Udq3uegcLp56giLN5UMRKYlMxfC/u77cwgimTCB4Oi8X7SXS2UKwhRZWq9WJrUptN6jNGwvFx2W62f7PmhcZaupWtDVge7z1hAqikE88rfEEiItnexBfC2ABeeDxBA4TIQQURfN84FeoueBDqo447P68pzv2i5LpgK7EGuG4ngydGwvuJEwlxzL9keJgnqKQYcHMtCcDEvCeBwWikKgnAAEpmAiCgLEsmBGhsPCkAQmiiXMADW3gxC2MRcOCWkxPiGz4hoZn4l44L6JwkhSXpcmcLwCggL4ukyaJpBwLASA6nQUTkJQoX0NExAEuCvhYAAbngywAGp4JgADuADy8SMJw2k0LQ%2BHEN5OHueEQS1PKBW8JVzDEPK2XhNoFR%2BdprJsII2UMLQNX%2BaQWAsIYwDiAN%2BB6pUiWYN5A2YKoFQmPhtXkIIrTubQeDhHKjUuFg7lHngLC1XMVAGMACjpVluX5dwvD8IIIhiOwUgyIIigqOoA26Fw%2BgjSApjmPoW3eZAcyoPE7SzT8ThPBDBA/PQ01gSiCi/D8QzAIxvoEAgESJfiLA/IcwQ/JgTCVejV5Gl4N6Kkq6xeF5rRtakDgMM4riNHoARBL0xT9L9OQpAIIyeELSQiwwkx9NEv3lJUAidMMXOZPLLOKx0QzdHzUyC4MXRi3o4x1DLAty3MqmLMsEhiRJbkDfJHBLspyBPFwLb4uCLYaE8EC4IQJCaVwMy8H5WgzHMCDk1g0RfKQRleCZGxeFiWIpwAnDEkh2eCGfgvizkcK5pDHVwGi%2BNJslO15Pk6XpcxBYgKCpvEYVkBQEBRf0sVewlmDJWlGU5Xl0mFXQJVldaA31dVK2z41zWtTYK2dYwBA9X17lDSNY2yRNrPTbNsnzYty13at%2BHiQNm3bdVe0rLJh3HXdp3nZdw83WP92yE94ivQ9eQSg1DuV0BsP6RgAbKWBuEUG8d4apGhrDeGiMB6tF9GjH4GNahYwIDjPGxQCZExJpgMmFNGxYOpozOmMZlSM2Zm0NmEBHBG1%2BrzQoss9DC3aKwhIkt2hm2mOrRhStta8IVu0ZWOsOHm2NmI1W4sDam11pwkO8xrYvTthwSSpAq68Cdi7cwbsPZex9n7AORB6zrC0qHeu/lI4J28JIFsXAYgbCxGZHOGcsQZw0BsDO4Dr4lzLhXXR7ka6WDruHAKTcIAhVbu3CKXcEnRQBnFfug9MBXRHrdcexUohTwqlVRq89ilNRam1VeqYuob16v1femBhpGD3rwA%2BU0ZruVPsgJaKxtJBCvhtLaO15QPwOsQI6J0%2BDv2yV/FagC/4vWkIAj6IDvreAgcYaBt84Hg0hogzgMM4aQ1QcjDB6NMbYxRLjfGhNiYCFIeTSmlCpQ0xoSVOhXgmYtBEZ4ZhHNeHsP5kIvhuRUi8O4akQR%2BsJFVHkRkRRMLRETBUbI%2BWcLuZouRTIoRls1I2zUdfHReiPLO0BgQYxntva%2B39vgSxwdbHRIcYnOIQSHbV08pE3yDctFfOJRExlcxpqlTZpIIAA%3D) but I like it.

**Review by ithinuel — APPROVED** [review]
I fixed a little issue with my previous suggestion and renamed the example.
Tested on hardware, work nice :)

**Review by 9names — APPROVED** [review]
LGTM

**Pull request merged**